homeassistant/components/denonavr/__init__.py

```diff
@@ -56,11 +55,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         raise ConfigEntryNotReady from ex
     receiver = connect_denonavr.receiver
 
-    undo_listener = entry.add_update_listener(update_listener)
+    entry.async_on_unload(entry.add_update_listener(update_listener))
 
     hass.data[DOMAIN][entry.entry_id] = {
         CONF_RECEIVER: receiver,
-        UNDO_UPDATE_LISTENER: undo_listener,
```

Comment:
I have kept the existing dict structure for now - it can be removed in a follow-up PR when it is migrated to runtime_data